test/libyul/ControlFlowSideEffectsTest.cpp

```diff
@@ -57,19 +59,21 @@ ControlFlowSideEffectsTest::ControlFlowSideEffectsTest(std::string const& _filen
 
 TestCase::TestResult ControlFlowSideEffectsTest::run(std::ostream& _stream, std::string const& _linePrefix, bool _formatted)
 {
-	auto const& dialect = CommonOptions::get().evmDialect();
-	Object obj;
-	auto parsingResult = parse(m_source);
-	obj.setCode(parsingResult.first, parsingResult.second);
-	if (!obj.hasCode())
-		BOOST_THROW_EXCEPTION(std::runtime_error("Parsing input failed."));
+	YulStack yulStack = parseYul(m_source);
+	solUnimplementedAssert(yulStack.parserResult()->subObjects.empty(), "Tests with subobjects not supported.");
+
+	if (yulStack.hasErrors())
+	{
+		printYulErrors(yulStack, _stream, _linePrefix, _formatted);
+		return TestResult::FatalError;
+	}
 
 	ControlFlowSideEffectsCollector sideEffects(
-		dialect,
-		obj.code()->root()
+		*yulStack.parserResult()->dialect(),
```

Comment:
Generally I think it would be cool if the stack would already do the parsing upon construction so that all methods are readily usable once the its been instantiated

Reply:
Ah wait, I do see one reason not to do parsing immediately - this would make collecting parsing errors more annoying (and different from other stages). Currently they end up on the error list held by the stack. If constructor fails, we won't have that list there so we'd have to instead pass it in from the outside.

Reply:
> Ah wait, I do see one reason not to do parsing immediately - this would make collecting parsing errors more annoying (and different from other stages). Currently they end up on the error list held by the stack. If constructor fails, we won't have that list there so we'd have to instead pass it in from the outside.

Ahhh I see, I didn't think about that. Makes sense, thanks for clarifying :)

Reply:
`YulStack::dialect()` added.